OIPA/api/transaction/serializers.py

```diff
@@ -310,10 +310,10 @@ def update(self, instance, validated_data):
         update_instance.id = instance.id
         update_instance.save()
 
+
         if provider_data.get('ref'):
-            provider_org = models.TransactionProvider.objects.create(
```

Comment:
I think there's 2 scenario's here:

-The user already created a transaction with the provider-org set. (covered by functionality after pull request)
-The user already created a transaction without setting a provider-org, the user later edits the transaction and adds the provider-org (covered by functionality before pull request)

So I think you need to check if the provider exists, if not create else update?